test/hotspot/gtest/x86/x86-asmtest.py

```diff
@@ -0,0 +1,706 @@
+    ],
+}
+
+instruction_set64 = {
+    TwoRegInstruction: [
+        ('adcq', 'adc', 64),
+        ('imulq', 'imul', 64),
+        ('popcntq', 'popcnt', 64),
+        ('sbbq', 'sbb', 64),
+        ('subq', 'sub', 64),
+        ('tzcntq', 'tzcnt', 64),
+        ('lzcntq', 'lzcnt', 64),
+        ('addq', 'add', 64),
+        ('andq', 'and', 64),
+        ('orq', 'or', 64),
+        ('xorq', 'xor', 64)
+    ],
+    MemRegInstruction: [
+        ('addq', 'add', 64),
+        ('andq', 'and', 64),
+        ('orq', 'or', 64),
+        ('xorq', 'xor', 64),
+        ('subq', 'sub', 64)
+    ],
+    MemImmInstruction: [
+        ('andq', 'and', 64),
+        ('addq', 'add', 64),
+        ('sarq', 'sar', 64),
+        ('sbbq', 'sbb', 64),
+        ('shrq', 'shr', 64),
+        ('subq', 'sub', 64),
+        ('xorq', 'xor', 64),
+        ('orq', 'or', 64)
+    ],
+    RegMemInstruction: [
+        ('addq', 'add', 64),
+        ('andq', 'and', 64),
+        ('lzcntq', 'lzcnt', 64),
+        ('orq', 'or', 64),
+        ('adcq', 'adc', 64),
+        ('imulq', 'imul', 64),
+        ('popcntq', 'popcnt', 64),
+        ('sbbq', 'sbb', 64),
+        ('subq', 'sub', 64),
+        ('tzcntq', 'tzcnt', 64),
+        ('xorq', 'xor', 64)
+    ],
+    RegImmInstruction: [
+        ('addq', 'add', 64),
+        ('andq', 'and', 64),
+        ('adcq', 'adc', 64),
+        ('rclq', 'rcl', 64),
+        ('rcrq', 'rcr', 64),
+        ('rolq', 'rol', 64),
+        ('rorq', 'ror', 64),
+        ('sarq', 'sar', 64),
+        ('sbbq', 'sbb', 64),
+        ('shlq', 'shl', 64),
+        ('shrq', 'shr', 64),
+        ('subq', 'sub', 64),
+        ('xorq', 'xor', 64),
+    ],
+    CondRegMemInstruction: [
+        ('cmovq', 'cmov', 64, key) for key in cond_to_suffix.keys()
+    ],
+    RegInstruction: [
+        ('divq', 'div', 64),
+        ('idivq', 'idiv', 64),
+        ('imulq', 'imul', 64),
+        ('mulq', 'mul', 64),
+        ('negq', 'neg', 64),
+        ('notq', 'not', 64),
+        ('rolq', 'rol', 64),
+        ('rorq', 'ror', 64),
+        ('sarq', 'sar', 64),
+        ('shlq', 'shl', 64),
+        ('shrq', 'shr', 64),
+        ('incrementq', 'inc', 64),
+        ('decrementq', 'dec', 64)
+    ],
+    MemInstruction: [
+        ('mulq', 'mul', 64),
+        ('negq', 'neg', 64),
+        ('sarq', 'sar', 64),
+        ('shrq', 'shr', 64),
+        ('incrementq', 'inc', 64),
+        ('decrementq', 'dec', 64)
+    ],
+    RegMemImmInstruction: [
+        ('imulq', 'imul', 64)
+    ],
+    RegRegImmInstruction: [
+        ('imulq', 'imul', 64),
+        ('shldq', 'shld', 64),
+        ('shrdq', 'shrd', 64)
+    ],
+    RegImm32Instruction: [
+        ('orq_imm32', 'or', 64),
+        ('subq_imm32', 'sub', 64)
+    ],
+    Pop2Instruction: [
+        ('pop2', 'pop2', 64),
+        ('pop2p', 'pop2p', 64)
+    ],
+    Push2Instruction: [
+        ('push2', 'push2', 64),
+        ('push2p', 'push2p', 64)
+    ],
+}
+
+for RegOp, ops in instruction_set.items():
```

Comment:
Rest of the code is modular, can you kindly refactor following code into a top level routine called from __main__ method.

Reply:
Binutils toolset is specfic for linux. Please add 
relevant OS.name check and exit

Reply:
Thanks, done.